parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -70,135 +69,158 @@ pub enum Destination {
 	ForeignAccountId20 { para_id: u32, id: [u8; 20] },
 }
 
-impl From<MessageV1> for Xcm<()> {
-	fn from(val: MessageV1) -> Self {
-		val.message.convert(val.chain_id)
-	}
+pub struct VersionedMessageToXcmConverter<G> {
```

Comment:
This struct name is a bit too verbose. Let's shorten it. And rename the generic parameter `G` to `CreateAssetCall`.

```suggestion
pub struct MessageToXcm<CreateAssetCall> {
```